### ISSUE TITLE
RFC: Should the language be javascript (long) or js (short)?

### ISSUE THREAD

**Issue by ben-elttam** [returntocorp/sgrep-rules#2754]
**Describe the bug**

Semgrep rules specify the language(s), and this can be either long form (e.g. `javascript`), or short form (e.g. `js`).

Which should be preferred?
(Considering: Linting rules / checks in CI could be advising one over the other).

**To Reproduce**

Write a rule, similar to one of the many rules like `javascript/lang/best-practice/assigned-undefined.yaml`

```yaml
rules:
  - id: my-new-rule
    languages:
      - javascript
      - typescript
```

Run semgrep (meta) against the rule for linting.

`yaml/semgrep/consistency/lang-consistency-js.yaml` complains the language should be `js` not `javascript`.

> Found 'javascript' in language config which diverges from semgrep.dev normalization.
    Please use 'js' instead.

**Expected behavior**

Consistency between the registry rules (returntocorp/semgrep-rules) and the `yaml/semgrep/consistency/lang-consistency-*` rules.

The consistency rules prefer short form, but the majority of the registry uses the long form.

Looking at returntocorp/semgrep-rules there are 17 `js` vs 314 `javascript` rules and 18 `ts` vs `207` typescript rules.

This was calculated using the snippet:

```
for x in $(fd '\.ya?ml$' --exclude  '*.test.*' ); do yq '.rules | map(.languages) | flatten' "$x"; done | jq -s 'flatten | group_by(.) | map({"\(first)": length}) | add'
```

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [ ] P2: annoying but not blocking me
- [X] P3: What does return2corp and the community think?

**Additional context**

As a community poll, please vote for long form by clicking the 🚀 and the short form by clicking the 🎉

**Comment by LewisArdern** [comment]
My personal vote (and Kurts) would be Long, you are welcome to open PRs and fix known ones which use shorthand, you can add me to the commit and i'll approve.

I will create a linear ticket internally to 'clean up' and enforce long format eventually, but its not something we will be prioritizing this quarter

**Comment by ben-elttam** [comment]
Maybe needs a low priority label?
